agent/my_daemon.py

```diff
@@ -39,13 +42,50 @@ def set_status(processes, api_key=None):
             os.makedirs(os.path.dirname(filename))
         try:
             with open(filename, 'w+') as stream:
-                config = yaml.load(stream)
-                if not config:
-                    config = {}
-                    if api_key:
-                        config['api_key'] = api_key
+                stream.truncate(0)
+                config = {'api_key': api_key}
                 for process in processes:
                     config[process.process.name()] = process.process.ppid()
                 yaml.dump(config, stream)
         except FileNotFoundError:
             logger.warning("No daemon configuration file found")
+
+    @staticmethod
+    def update_processes(processes_name, processes_pid):
+        processes = []
+        if processes_name:
+            for process_name in processes_name:
+                processes.append(
+                    Process.create_name_resource(process_name))
+        if processes_pid:
+            for resource_pid in processes_pid:
+                processes.append(
+                    Process.create_pid_resource(resource_pid))
```

Comment:
`processes = [Process.create_pid_resource(resource_pid) for resource_pid in processes_pid]` one liner